## ORGANS: Liver Import + Export Flow

This notebook contains analysis of the import and export flow of livers between states.

---

### Initialize environment

In [2]:
library(tidyverse)

In [3]:
root_path     <- here::here()
data_path     <- paste0(root_path, "/01_data")
analysis_path <- paste0(root_path, "/02_analysis")

In [4]:
setwd(data_path)

transplant <- readRDS("clean/transplant_clean.rds")

setwd(analysis_path)

---

### Calculate number of livers coming into and going out of each state

In [5]:
txByYearDonCanState <- transplant %>%
  filter(REC_TX_DT_YEAR %in% 2018:2021) %>%
  # excluding 1701 transplants with missing donor information
  tidylog::filter(!is.na(DON_HOSPITAL_STATE)) %>%
  # excluding 3 transplants with missing candidate information
  tidylog::filter(!is.na(CAN_HOSPITAL_STATE)) %>%
  count(REC_TX_DT_YEAR, DON_HOSPITAL_STATE, CAN_HOSPITAL_STATE, name = "N_TX") %>%
  group_by(REC_TX_DT_YEAR, DON_HOSPITAL_STATE) %>%
  mutate(PCT_TX_DON_STATE = N_TX / sum(N_TX)) %>%
  ungroup() %>%
  group_by(REC_TX_DT_YEAR, CAN_HOSPITAL_STATE) %>%
  mutate(PCT_TX_CAN_STATE = N_TX / sum(N_TX)) %>%
  ungroup() %>%
  arrange(REC_TX_DT_YEAR, DON_HOSPITAL_STATE, -N_TX)

write.csv(x = txByYearDonCanState,
          file = "output/tables/table3-tx_by_year_donstate_canstate.csv",
          row.names = F)

txByYearDonCanState %>% head()

filter: removed 1,701 rows (5%), 31,468 rows remaining

filter: removed 3 rows (<1%), 31,465 rows remaining



REC_TX_DT_YEAR DON_HOSPITAL_STATE CAN_HOSPITAL_STATE N_TX PCT_TX_DON_STATE
1 2018           AK                 WA                 15   0.78947368      
2 2018           AK                 OR                  3   0.15789474      
3 2018           AK                 AZ                  1   0.05263158      
4 2018           AL                 AL                 85   0.68000000      
5 2018           AL                 FL                 17   0.13600000      
6 2018           AL                 GA                  8   0.06400000      
  PCT_TX_CAN_STATE
1 0.106382979     
2 0.036585366     
3 0.003571429     
4 0.714285714     
5 0.032882012     
6 0.030534351

In [6]:
importExportInternalTxByYearState <- bind_rows(
  txByYearDonCanState %>%
    filter(DON_HOSPITAL_STATE == CAN_HOSPITAL_STATE) %>%
    select(YEAR = REC_TX_DT_YEAR, STATE = DON_HOSPITAL_STATE, N = N_TX) %>%
    mutate(TYPE = "N_INTERNAL_LIVERS"),
  txByYearDonCanState %>%
    filter(DON_HOSPITAL_STATE != CAN_HOSPITAL_STATE) %>%
    count(YEAR = REC_TX_DT_YEAR, STATE = DON_HOSPITAL_STATE, wt = N_TX, name = "N") %>%
    mutate(TYPE = "N_EXPORTED_LIVERS"),
  txByYearDonCanState %>%
    filter(DON_HOSPITAL_STATE != CAN_HOSPITAL_STATE) %>%
    count(YEAR = REC_TX_DT_YEAR, STATE = CAN_HOSPITAL_STATE, wt = N_TX, name = "N") %>%
    mutate(TYPE = "N_IMPORTED_LIVERS")) %>%
  pivot_wider(id_cols = c("YEAR", "STATE"), names_from = TYPE, values_from = N) %>%
  mutate(across(matches("^N_"), replace_na, 0)) %>%
  mutate(MAJORITY_EXPORTER = N_EXPORTED_LIVERS > N_INTERNAL_LIVERS,
         MAJORITY_IMPORTER = N_IMPORTED_LIVERS > N_INTERNAL_LIVERS) %>%
  arrange(YEAR, STATE)

write.csv(x = importExportInternalTxByYearState,
          file = "output/tables/table3-tx_import_export_internal_by_year_state.csv",
          row.names = F)

importExportInternalTxByYearState %>% head()

YEAR STATE N_INTERNAL_LIVERS N_EXPORTED_LIVERS N_IMPORTED_LIVERS
1 2018 AK      0                19                 0              
2 2018 AL     85                40                34              
3 2018 AR     27                23                10              
4 2018 AZ    112                60               168              
5 2018 CA    566               115               219              
6 2018 CO     61                24                46              
  MAJORITY_EXPORTER MAJORITY_IMPORTER
1  TRUE             FALSE            
2 FALSE             FALSE            
3 FALSE             FALSE            
4 FALSE              TRUE            
5 FALSE             FALSE            
6 FALSE             FALSE

In [7]:
importExportInternalTxByYearState %>%
  group_by(YEAR) %>%
  summarise(across(matches("^N_"), sum, na.rm = T)) %>%
  mutate(PCT_IMPORTED_LIVERS = N_IMPORTED_LIVERS / (N_INTERNAL_LIVERS + N_IMPORTED_LIVERS),
         PCT_EXPORTED_LIVERS = N_EXPORTED_LIVERS / (N_INTERNAL_LIVERS + N_EXPORTED_LIVERS))

YEAR N_INTERNAL_LIVERS N_EXPORTED_LIVERS N_IMPORTED_LIVERS
1 2018 5014              2334              2334             
2 2019 5378              2522              2522             
3 2020 3856              4122              4122             
4 2021 3717              4522              4522             
  PCT_IMPORTED_LIVERS PCT_EXPORTED_LIVERS
1 0.3176375           0.3176375          
2 0.3192405           0.3192405          
3 0.5166708           0.5166708          
4 0.5488530           0.5488530

In [8]:
importExportShift <- importExportInternalTxByYearState %>%
  mutate(PCT_IMPORT = N_IMPORTED_LIVERS / (N_INTERNAL_LIVERS + N_IMPORTED_LIVERS),
         PCT_EXPORT = N_EXPORTED_LIVERS / (N_INTERNAL_LIVERS + N_EXPORTED_LIVERS)) %>%
  filter(YEAR %in% c(2019, 2021)) %>%
  pivot_wider(id_cols = STATE, names_from = YEAR, values_from = matches("PCT_")) %>%
  mutate(PCT_IMPORT_SHIFT = PCT_IMPORT_2021 - PCT_IMPORT_2019, .after = PCT_IMPORT_2021) %>%
  mutate(PCT_EXPORT_SHIFT = PCT_EXPORT_2021 - PCT_EXPORT_2019, .after = PCT_EXPORT_2021)
 

print("Summary of PCT_IMPORT_SHIFT")
summary(importExportShift$PCT_IMPORT_SHIFT)

importExportShift %>%
  filter(between(PCT_IMPORT_SHIFT, 0.24, 0.25))

print("Summary of PCT_EXPORT_SHIFT")
summary(importExportShift$PCT_EXPORT_SHIFT)

importExportShift %>%
  filter(between(PCT_EXPORT_SHIFT, 0.185, 0.205))

importExportShift

[1] "Summary of PCT_IMPORT_SHIFT"


    Min.  1st Qu.   Median     Mean  3rd Qu.     Max.     NA's 
-0.05195  0.11143  0.24066  0.24470  0.34964  0.62076       14 

STATE PCT_IMPORT_2019 PCT_IMPORT_2021 PCT_IMPORT_SHIFT PCT_EXPORT_2019
1 TN    0.3185841       0.5592417       0.2406576        0.4166667      
  PCT_EXPORT_2021 PCT_EXPORT_SHIFT
1 0.7066246       0.2899579

[1] "Summary of PCT_EXPORT_SHIFT"


   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
-0.2707  0.0000  0.1959  0.1811  0.3393  0.4798       1 

STATE PCT_IMPORT_2019 PCT_IMPORT_2021 PCT_IMPORT_SHIFT PCT_EXPORT_2019
1 CO    0.4365079       0.5447761       0.1082682        0.3106796      
2 CT    0.6136364       0.8823529       0.2687166        0.7118644      
  PCT_EXPORT_2021 PCT_EXPORT_SHIFT
1 0.5000000       0.1893204       
2 0.9142857       0.2024213

STATE PCT_IMPORT_2019 PCT_IMPORT_2021 PCT_IMPORT_SHIFT PCT_EXPORT_2019
1  AK           NaN             NaN              NaN      1.00000000     
2  AL    0.28125000      0.56944444       0.28819444      0.36986301     
3  AR    0.32558140      0.64444444       0.31886305      0.50847458     
4  AZ    0.63496933      0.66559486       0.03062553      0.39593909     
5  CA    0.27972028      0.29502762       0.01530734      0.16486486     
6  CO    0.43650794      0.54477612       0.10826818      0.31067961     
7  CT    0.61363636      0.88235294       0.26871658      0.71186441     
8  DC    0.80000000      0.91743119       0.11743119      0.40625000     
9  DE           NaN             NaN              NaN      1.00000000     
10 FL    0.28160000      0.25662483      -0.02497517      0.23247863     
11 GA    0.29615385      0.58545455       0.28930070      0.27952756     
12 HI    0.07692308      0.05882353      -0.01809955      0.42857143     
13 IA    0.21875000      0.33333333       0.11458333      0.53703704     
14 ID           NaN             NaN              NaN      1.00000000     
15 IL    0.20547945      0.60747664       0.40199718      0.22666667     
16 IN    0.27777778      0.67114094       0.39336316      0.35403727     
17 KS    0.63750000      0.74000000       0.10250000      0.63750000     
18 KY    0.35820896      0.79569892       0.43748997      0.47560976     
19 LA    0.40585774      0.51149425       0.10563651      0.24064171     
20 MA    0.57661290      0.80800000       0.23138710      0.15322581     
21 MD    0.41242938      0.74285714       0.33042776      0.28275862     
22 ME           NaN             NaN              NaN      1.00000000     
23 MI    0.15121951      0.61951220       0.46829268      0.21973094     
24 MN    0.54088050      0.69339623       0.15251572      0.32407407     
25 MO    0.26136364      0.51219512       0.25083149      0.48818898     
26 MS    0.29411765      0.50000000       0.20588235      0.43750000     
27 MT           NaN             NaN              NaN      1.00000000     
28 NC    0.26229508      0.72429907       0.46200398      0.28286853     
29 ND           NaN             NaN              NaN      1.00000000     
30 NE    0.62352941      0.68817204       0.06464263      0.21951220     
31 NH           NaN             NaN              NaN      1.00000000     
32 NJ    0.12500000      0.74576271       0.62076271      0.69565217     
33 NM           NaN             NaN              NaN      1.00000000     
34 NV           NaN             NaN              NaN      1.00000000     
35 NY    0.21158690      0.71119134       0.49960443      0.06006006     
36 OH    0.32629108      0.63312369       0.30683261      0.11963190     
37 OK    0.37500000      0.67200000       0.29700000      0.63157895     
38 OR    0.51315789      0.69879518       0.18563729      0.24489796     
39 PA    0.36105477      0.71046229       0.34940752      0.22413793     
40 PR    0.14285714      0.09090909      -0.05194805      0.57647059     
41 RI           NaN             NaN              NaN      1.00000000     
42 SC    0.13333333      0.73239437       0.59906103      0.47580645     
43 SD    0.66666667      0.83333333       0.16666667      0.81818182     
44 TN    0.31858407      0.55924171       0.24065764      0.41666667     
45 TX    0.07891332      0.25801953       0.17910620      0.08365508     
46 UT    0.42857143      0.63709677       0.20852535      0.34328358     
47 VA    0.46703297      0.81690141       0.34986844      0.40853659     
48 VI           NaN              NA               NA      1.00000000     
49 VT           NaN             NaN              NaN      1.00000000     
50 WA    0.28571429      0.30147059       0.01575630      0.31818182     
51 WI    0.26206897      0.72340426       0.46133529      0.30519481     
52 WV           NaN             NaN              NaN      1.00000000     
53 WY           NaN             NaN              NaN      1.00000000     
   PCT_

In [9]:
majorityStatusByYear <- importExportInternalTxByYearState %>%
  group_by(YEAR) %>%
  summarise(across(matches("MAJORITY"), ~ sum(as.numeric(.x))))

write.csv(x = majorityStatusByYear,
          file = "output/tables/table3-majority_import_export_states_by_year.csv",
          row.names = F)

majorityStatusByYear

YEAR MAJORITY_EXPORTER MAJORITY_IMPORTER
1 2018 22                 7               
2 2019 22                 9               
3 2020 38                30               
4 2021 42                31

---

### Re-run same calculations, but using UNOS 730-day periods

In [10]:
unos_periods <- list(pre  = as.Date(c("2018-02-03", "2020-02-03")),
                     post = as.Date(c("2020-02-04", "2022-02-03")))

In [11]:
txByYearDonCanState.v2 <- transplant %>%
  filter(between(REC_TX_DT_DATE, unos_periods$pre[1], unos_periods$post[2])) %>%
  tidylog::filter(!is.na(CAN_HOSPITAL_STATE), !is.na(DON_HOSPITAL_STATE)) %>%
  mutate(PERIOD = case_when(between(REC_TX_DT_DATE, unos_periods$pre[1], unos_periods$pre[2]) ~ "PRE",
                            between(REC_TX_DT_DATE, unos_periods$post[1], unos_periods$post[2]) ~ "POST")) %>%
  count(PERIOD, DON_HOSPITAL_STATE, CAN_HOSPITAL_STATE, name = "N_TX") %>%
  group_by(PERIOD, DON_HOSPITAL_STATE) %>%
  mutate(PCT_TX_DON_STATE = N_TX / sum(N_TX)) %>%
  ungroup() %>%
  group_by(PERIOD, CAN_HOSPITAL_STATE) %>%
  mutate(PCT_TX_CAN_STATE = N_TX / sum(N_TX)) %>%
  ungroup() %>%
  arrange(PERIOD, DON_HOSPITAL_STATE, -N_TX)

txByYearDonCanState.v2 %>% head()

filter: removed 1,722 rows (5%), 31,487 rows remaining



PERIOD DON_HOSPITAL_STATE CAN_HOSPITAL_STATE N_TX PCT_TX_DON_STATE
1 POST   AK                 WA                 21   0.55263158      
2 POST   AK                 OR                 12   0.31578947      
3 POST   AK                 CA                  3   0.07894737      
4 POST   AK                 AZ                  2   0.05263158      
5 POST   AL                 GA                 65   0.20376176      
6 POST   AL                 AL                 60   0.18808777      
  PCT_TX_CAN_STATE
1 0.078066914     
2 0.074074074     
3 0.001645639     
4 0.003333333     
5 0.125000000     
6 0.370370370

In [12]:
importExportInternalTxByYearState.v2 <- bind_rows(
  txByYearDonCanState.v2 %>%
    filter(DON_HOSPITAL_STATE == CAN_HOSPITAL_STATE) %>%
    select(PERIOD, STATE = DON_HOSPITAL_STATE, N = N_TX) %>%
    mutate(TYPE = "N_INTERNAL_LIVERS"),
  txByYearDonCanState.v2 %>%
    filter(DON_HOSPITAL_STATE != CAN_HOSPITAL_STATE) %>%
    count(PERIOD, STATE = DON_HOSPITAL_STATE, wt = N_TX, name = "N") %>%
    mutate(TYPE = "N_EXPORTED_LIVERS"),
  txByYearDonCanState.v2 %>%
    filter(DON_HOSPITAL_STATE != CAN_HOSPITAL_STATE) %>%
    count(PERIOD, STATE = CAN_HOSPITAL_STATE, wt = N_TX, name = "N") %>%
    mutate(TYPE = "N_IMPORTED_LIVERS")) %>%
  pivot_wider(id_cols = c("PERIOD", "STATE"), names_from = TYPE, values_from = N) %>%
  mutate(across(matches("^N_"), replace_na, 0)) %>%
  mutate(MAJORITY_EXPORTER = N_EXPORTED_LIVERS > N_INTERNAL_LIVERS,
         MAJORITY_IMPORTER = N_IMPORTED_LIVERS > N_INTERNAL_LIVERS) %>%
  arrange(STATE, PERIOD)

importExportInternalTxByYearState.v2 %>% head()

PERIOD STATE N_INTERNAL_LIVERS N_EXPORTED_LIVERS N_IMPORTED_LIVERS
1 POST   AK      0                38                 0              
2 PRE    AK      0                46                 0              
3 POST   AL     60               259               102              
4 PRE    AL    177                95                68              
5 POST   AR     27                97                50              
6 PRE    AR     57                51                24              
  MAJORITY_EXPORTER MAJORITY_IMPORTER
1  TRUE             FALSE            
2  TRUE             FALSE            
3  TRUE              TRUE            
4 FALSE             FALSE            
5  TRUE              TRUE            
6 FALSE             FALSE

In [13]:
importExportInternalTxByYearState.v2 %>%
  group_by(PERIOD) %>%
  summarise(across(matches("^N_"), sum, na.rm = T)) %>%
  mutate(PCT_IMPORTED_LIVERS = N_IMPORTED_LIVERS / (N_INTERNAL_LIVERS + N_IMPORTED_LIVERS),
         PCT_EXPORTED_LIVERS = N_EXPORTED_LIVERS / (N_INTERNAL_LIVERS + N_EXPORTED_LIVERS))

PERIOD N_INTERNAL_LIVERS N_EXPORTED_LIVERS N_IMPORTED_LIVERS
1 POST    7365             8803              8803             
2 PRE    10447             4872              4872             
  PCT_IMPORTED_LIVERS PCT_EXPORTED_LIVERS
1 0.5444706           0.5444706          
2 0.3180364           0.3180364

Using the 730-day periods, we're getting effectively the same pre- and post-policy shifts in national share of livers imported/exported (shifting from 31.8% to 54.4% using 730-days; shifting from 31.9% to 54.9% using 2019 vs. 2021)

Seeing similar results in next cell: the median import shift by state was +26.7pp (as compared to +24pp above); the median export shift by state was +20.8pp (as compared to 19.6pp above).

In [14]:
importExportShift.v2 <- importExportInternalTxByYearState.v2 %>%
  mutate(PCT_IMPORT = N_IMPORTED_LIVERS / (N_INTERNAL_LIVERS + N_IMPORTED_LIVERS),
         PCT_EXPORT = N_EXPORTED_LIVERS / (N_INTERNAL_LIVERS + N_EXPORTED_LIVERS)) %>%
  pivot_wider(id_cols = STATE, names_from = PERIOD, values_from = matches("PCT_")) %>%
  mutate(PCT_IMPORT_SHIFT = PCT_IMPORT_POST - PCT_IMPORT_PRE, .after = PCT_IMPORT_PRE) %>%
  mutate(PCT_EXPORT_SHIFT = PCT_EXPORT_POST - PCT_EXPORT_PRE, .after = PCT_EXPORT_PRE)
 
print("Summary of PCT_IMPORT_SHIFT")
summary(importExportShift.v2$PCT_IMPORT_SHIFT)

print("Summary of PCT_EXPORT_SHIFT")
summary(importExportShift.v2$PCT_EXPORT_SHIFT)

importExportShift.v2

[1] "Summary of PCT_IMPORT_SHIFT"


   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
-0.0522  0.1588  0.2670  0.2533  0.3552  0.6433      14 

[1] "Summary of PCT_EXPORT_SHIFT"


   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
-0.2583  0.0000  0.2084  0.1877  0.3190  0.5598       1 

STATE PCT_IMPORT_POST PCT_IMPORT_PRE PCT_IMPORT_SHIFT PCT_EXPORT_POST
1  AK           NaN             NaN             NaN      1.0000000      
2  AL    0.62962963      0.27755102      0.35207861      0.8119122      
3  AR    0.64935065      0.29629630      0.35305435      0.7822581      
4  AZ    0.69500000      0.62418301      0.07081699      0.5390428      
5  CA    0.30005485      0.27801592      0.02203893      0.1175657      
6  CO    0.46747967      0.41949153      0.04798815      0.4842520      
7  CT    0.92929293      0.61458333      0.31470960      0.9457364      
8  DC    0.92647059      0.76683938      0.15963121      0.6875000      
9  DE           NaN             NaN             NaN      1.0000000      
10 FL    0.25205071      0.26637931     -0.01432860      0.2039683      
11 GA    0.62307692      0.31428571      0.30879121      0.5975359      
12 HI    0.06250000      0.04166667      0.02083333      0.1666667      
13 IA    0.41304348      0.23437500      0.17866848      0.7938931      
14 ID           NaN             NaN             NaN      1.0000000      
15 IL    0.57335329      0.18650089      0.38685241      0.5136519      
16 IN    0.64150943      0.28424658      0.35726286      0.7665848      
17 KS    0.71578947      0.61744966      0.09833981      0.8689320      
18 KY    0.81927711      0.34027778      0.47899933      0.8728814      
19 LA    0.51111111      0.44672131      0.06438980      0.5331565      
20 MA    0.78642715      0.54297694      0.24345021      0.4702970      
21 MD    0.72795970      0.46091644      0.26704326      0.5970149      
22 ME           NaN             NaN             NaN      1.0000000      
23 MI    0.60904255      0.14676617      0.46227638      0.6776316      
24 MN    0.72906404      0.56369427      0.16536977      0.3604651      
25 MO    0.48041775      0.28125000      0.19916775      0.6098039      
26 MS    0.57647059      0.25263158      0.32383901      0.7482517      
27 MT           NaN             NaN             NaN      1.0000000      
28 NC    0.67654321      0.27937916      0.39716405      0.7380000      
29 ND           NaN             NaN             NaN      1.0000000      
30 NE    0.73469388      0.56000000      0.17469388      0.4800000      
31 NH           NaN             NaN             NaN      1.0000000      
32 NJ    0.78095238      0.13761468      0.64333770      0.9335260      
33 NM           NaN             NaN             NaN      1.0000000      
34 NV           NaN             NaN             NaN      1.0000000      
35 NY    0.72794118      0.21073298      0.51720819      0.5008432      
36 OH    0.62165059      0.30348259      0.31816800      0.5248991      
37 OK    0.67741935      0.27027027      0.40714908      0.6428571      
38 OR    0.70987654      0.46625767      0.24361887      0.7151515      
39 PA    0.68496420      0.35397317      0.33099103      0.6752768      
40 PR    0.06779661      0.12000000     -0.05220339      0.6604938      
41 RI           NaN             NaN             NaN      1.0000000      
42 SC    0.69047619      0.14864865      0.54182754      0.8388430      
43 SD    0.89473684      0.73684211      0.15789474      0.9200000      
44 TN    0.58574610      0.29530201      0.29044409      0.6920530      
45 TX    0.25100671      0.09047941      0.16052731      0.2051282      
46 UT    0.62184874      0.44805195      0.17379679      0.4339623      
47 VA    0.78536585      0.47687861      0.30848724      0.7317073      
48 VI            NA             NaN              NA             NA      
49 VT           NaN             NaN             NaN      1.0000000      
50 WA    0.34944238      0.30508475      0.04435763      0.4262295      
51 WI    0.66987179      0.30914826      0.36072353      0.6750789      
52 WV           NaN             NaN             NaN      1.0000000      
53 WY           NaN             NaN             NaN      1.0000000      
   PCT_EXPORT_PRE PCT_EXPORT_SHIFT
1  1.00000000      0.00000

In [15]:
majorityStatusByYear.v2 <- importExportInternalTxByYearState.v2 %>%
  group_by(PERIOD) %>%
  summarise(across(matches("MAJORITY"), ~ sum(as.numeric(.x))))

majorityStatusByYear.v2

PERIOD MAJORITY_EXPORTER MAJORITY_IMPORTER
1 POST   42                30               
2 PRE    21                 8

Again getting very similar results: a shift of 21 major exports in pre-policy period to 42 in post-policy period, as compared to 22 in FY2019 vs. 42 in FY2021 (see above)